### PR TITLE
Refactor the Maven project

### PR DESCRIPTION
**Pull request by yan-qi**
1. Changed the project structure for an easier deployment
2. Standardized the group ID and artifact ID 
3. Cleaned the POM files

Signed-off-by: Yan Qi yan.qi.asu@gmail.com

### CODE REVIEW

File: recordbuffers/pom.xml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+						</goals>
+					</execution>
+				</executions>
+			</plugin>
+			<plugin>
+				<groupId>com.mycila</groupId>
+				<artifactId>license-maven-plugin</artifactId>
+				<version>2.10</version>
+				<configuration>
+					<header>header_tmp.txt</header>
```
Comment (hzhang22) [inline recordbuffers/pom.xml]: Do we want to make a formal name for the license header?

**Comment by hzhang22** [comment]
The change looks good to me!

**Comment by yan-qi** [comment]
About the license the maven-plug-in can handle it for us.

**Comment by hzhang22** [comment]
That's pretty good now.

**Pull request merged**